backend/src/nodes/utils/ncnn_model.py

```diff
@@ -299,35 +299,35 @@ def __init__(
         self.blob_count: int = blob_count
         self.layer_list: List[NcnnLayer] = []
         self.weights_bin: bytes = b""
+        self.bin_length = 0
 
     @property
     def magic(self):
         return "7767517"
 
-    @staticmethod
-    def load_from_file(param_path: str = "", bin_path: str = "") -> "NcnnModel":
+    def load_from_file(self, param_path: str = "", bin_path: str = "") -> "NcnnModel":
```

Comment:
That was supposed to be a static constructor. You shouldn't call this method on anything by an empty NcnnModel anyway.

Reply:
Oh whoops, I meant to undo this once I finally was able to get the bin length. 

But, why is this supposed to be a static method? I don't really understand the reasoning

Reply:
A constructor is supposed to create a valid instance of a class. "Valid" here meaning "in a valid state." However, that's not what the NcnnModel constructor does. This constructor doesn't create a valid NCNN model. 

In other languages, you might make the empty constructor private and expose this method as a constructor overload, but Python supports neither of those features, so we have to resort to static constructor/factory methods instead.

Reply:
Why not just move this code to the constructor then?

Reply:
I believe we need the empty constructor somewhere else. Zamboni actually implemented "overloading" before moving to a static constructor method. It was really messy since we had to deduce what to do depending on which args we got. This might be doable for empty constructor + load from file, but it gets worse quickly. Just try adding another "overload"... 
There's also the problem that it's not terribly descriptive. Guess what `NcnnModel(someString)` does. `NcnnModel.load_from_file(someString)` tells you want it does.

For those reasons (and others), there are even some programming languages that don't do constructors at all. E.g. in Rust, there are only static constructors. This makes it really easy to understand what's going on. Ever spent 10mins reading through C++ docs, because your IDE couldn't figure out which of the 20+ vector overloads was called?

Reply:
makes sense